druid/examples/image.rs

```diff
@@ -12,44 +12,319 @@
-            .fix_width(100.0)
-            .center(),
-        1.0,
-    );
-
-    /*
-    // If you want to change the fill stratagy you can but you need the widget to be mut
-    let mut otherimage = Image::new(png_data);
-    otherimage.set_fill(FillStrat::FitWidth);
-    */
-
-    let otherimage = Image::new(png_data)
-        .fill_mode(FillStrat::FitWidth)
-        .border(Color::WHITE, 1.0);
-    col.add_flex_child(otherimage, 1.0);
-    col
+    // We create two images, one not having any modifications and the other
+    // forced to a fixed width, a fill strategy and an interpolation mode.
+    // You can see how this affects the final result. You can play with the
+    // interpolation mode to see how this affects things. (Note that this image
+    // is already anti-aliased so NearestNeighbor looks weird.)
+    Flex::column()
+        .with_flex_child(
+            Image::new(png_data.clone())
+                .fill_mode(FillStrat::FitWidth)
+                .interpolation_mode(InterpolationMode::NearestNeighbor)
+                .border(Color::WHITE, 1.0)
+                .fix_width(150.0)
+                .center(),
+            1.0,
+        )
+        .with_flex_child(
+            Image::new(png_data)
+                .border(Color::WHITE, 1.0),
+            1.0,
+        )
+}
+ */
+
+use druid::piet::InterpolationMode;
+use druid::widget::{prelude::*, FillStrat, Image};
+use druid::widget::{
+    Checkbox, CrossAxisAlignment, Flex, Label, RadioGroup, SizedBox, Stepper, TextBox, WidgetExt,
+};
+use druid::{AppLauncher, Color, Data, ImageBuf, Lens, LensExt, WindowDesc};
+
+const FILL_STRAT_OPTIONS: [(&str, FillStrat); 7] = [
+    ("Contain", FillStrat::Contain),
+    ("Cover", FillStrat::Cover),
+    ("Fill", FillStrat::Fill),
+    ("FitHeight", FillStrat::FitHeight),
+    ("FitWidth", FillStrat::FitWidth),
+    ("None", FillStrat::None),
+    ("ScaleDown", FillStrat::ScaleDown),
+];
+
+const INTERPOLATION_MODE_OPTIONS: [(&str, InterpolationModeData); 2] = [
+    (
+        "Bilinear",
+        InterpolationModeData::new(InterpolationMode::Bilinear),
+    ),
+    (
+        "NearestNeighbor",
+        InterpolationModeData::new(InterpolationMode::NearestNeighbor),
+    ),
+];
+
+#[derive(Clone, Data, Lens)]
+struct AppState {
+    demo_state: DemoState,
+    params: Params,
+}
+
+#[derive(Clone, Data, Lens)]
+struct DemoState {
+    pub input_text: String,
+    pub enabled: bool,
+    volume: f64,
+}
+
+#[derive(Clone, Data, Lens)]
+struct Params {
+    debug_layout: bool,
+    fill_strat: FillStrat,
+    interpolate: bool,
+    interpolation_mode: InterpolationModeData,
+    fix_width: bool,
+    width: f64,
+    fix_height: bool,
+    height: f64,
+}
+
+#[derive(Clone, PartialEq)]
+struct InterpolationModeData(InterpolationMode);
+
+impl InterpolationModeData {
+    const fn new(mode: InterpolationMode) -> InterpolationModeData {
+        InterpolationModeData(mode)
+    }
+}
+impl druid::Data for InterpolationModeData {
```

Comment:
we should be able to derive this, like:

```rust
#[derive(Clone, Data)]
 struct InterpolationModeData(
    #[data(same_fn="PartialEq::eq")]
    InterpolationMode
);
```

I would also be okay with us adding a `Data` impl to `InterpolationMode`, I did this for the various types in `examples/flex.rs` for this same reason.

Reply:
yes that would be great! I think it makes sense to have `InterpolationMode` in data for some things, its not something we want to prevent people from putting in there.

Reply:
Did we make this change? If we did we can also probably get rid of this `InterpolationModeData` type, and just use `InterpolationMode` directly?

Reply:
`InterpolationMode` does not have the data trait implemented for it, yet. It would be nice if that would be possible to do. 
I can create a PR for it later?

Reply:
@JAicewizard yes feel free to add it to this PR. Also if merge in the latest master it should fix that problem with CI failing.